src/web/app/components/feedback-path-panel/feedback-path-panel.component.ts

```diff
@@ -95,6 +97,22 @@ export class FeedbackPathPanelComponent {
     this.customFeedbackPath.emit(true);
   }
 
+  toggleSubMenu(menu: FeedbackParticipantType): void {
+    this.subMenuStatuses.set(menu, !this.subMenuStatuses.get(menu));
+  }
+
+  resetMenu(): void {
+    this.subMenuStatuses.forEach((_, key) => this.subMenuStatuses.set(key, false));
+  }
+
+  isSubMenuOpen(menu: FeedbackParticipantType): boolean {
+    let subMenuState: boolean | undefined = this.subMenuStatuses.get(menu);
+    if (subMenuState === undefined) {
+      subMenuState = false;
+    }
```

Comment:
Both approaches have their pros and cons, the current approach helps to make types more explicit. Will leave this as-is since the code itself is fine.